### PR TITLE
Make sure __all__ variables loaded in flow __init__.py

### PR DESCRIPTION
**Pull request by dschult**
The ```__all__``` variables are currently not loading correctly in the algorithms/flow/__init__.py file.
This fixes that.

### CODE REVIEW

File: networkx/algorithms/flow/__init__.py [hunk 1]
```diff
@@ -1,12 +1,6 @@
-from .maxflow import *
-from .mincost import *
-from .edmondskarp import *
-from .preflowpush import *
-from .shortestaugmentingpath import *
-from .capacityscaling import *
-from .networksimplex import *
-from .utils import build_flow_dict, build_residual_network
-
+# this import is used to combine the __all__ varibles.
```
Comment (NeilGirdhar) [inline networkx/algorithms/flow/__init__.py]: What's the point of lines 1–12?  If you just do the imports on lines 14–21, you don't need the effect of __all__, right?

Comment (dschult) [inline networkx/algorithms/flow/__init__.py]: The effect of ```__all__``` occurs when importing the flow module.
```from flow import *```
imports the items listed in the ```__all__``` variable defined in ```flow/__init__.py```
So we need an __all__ variable for the ```flow``` module (or maybe flow is a package...)

Anyway, sphinx doesn't build the docs correctly without it.

Comment (NeilGirdhar) [inline networkx/algorithms/flow/__init__.py]: I understand how `__all__` works, but you don't need to specify it if it lists everything in your module.

Comment (dschult) [inline networkx/algorithms/flow/__init__.py]: It doesn't work for sphinx.  try it.

Comment (NeilGirdhar) [inline networkx/algorithms/flow/__init__.py]: I believe you.  Sorry, edited my comment.

**Pull request merged**